src/internal/clusterstate/v2.4.0.go

```diff
@@ -24,6 +26,25 @@ var state_2_4_0 migrations.State = state_2_3_0.
 			return errors.Wrap(err, "could not create default project")
 		}
 		return nil
+	}).
+	Apply("Grant ProjectCreator to all users at cluster level", func(ctx context.Context, env migrations.Env) error {
+		key := auth_server.ResourceKey(&auth.Resource{Type: auth.ResourceType_CLUSTER})
+		rb := &auth.RoleBinding{Entries: make(map[string]*auth.Roles)}
+		if err := auth_server.RoleBindingCollection(nil, nil).ReadWrite(env.Tx).Get(key, rb); err != nil {
+			if errors.Is(err, col.ErrNotFound{}) {
+				// auth is not activated
+				return nil
+			}
+			return errors.EnsureStack(err)
+		}
+		if _, ok := rb.Entries[auth.AllClusterUsersSubject]; !ok {
+			rb.Entries[auth.AllClusterUsersSubject] = &auth.Roles{Roles: make(map[string]bool)}
+		}
+		rb.Entries[auth.AllClusterUsersSubject].Roles[auth.ProjectWriter] = true
```

Comment:
they should be assigned ProjectCreator instead of ProjectWriter at the cluster level. Since the default project is created at this point, we should also set allClusterUsers ProjectWriter for the default project